lib/ocpp/v201/charge_point.cpp

```diff
@@ -2624,56 +2592,59 @@ void ChargePoint::handle_change_availability_req(Call<ChangeAvailabilityRequest>
         return;
     }
 
+    // Check if we have any transaction running on the EVSE (or any EVSE if we're addressing the whole CS)
     const auto transaction_active = this->any_transaction_active(msg.evse);
+    // Check if we're already in the requested state
     const auto is_already_in_state = this->is_already_in_state(msg);
 
-    auto evse_id = 0; // represents the whole charging station
+    // evse_id will be 0 if we're addressing the whole CS, and >=1 otherwise
+    auto evse_id = 0;
     if (msg.evse.has_value()) {
         evse_id = msg.evse.value().id;
     }
 
     if (!transaction_active or is_already_in_state or
         (evse_id == 0 and msg.operationalStatus == OperationalStatusEnum::Operative)) {
+        // If the chosen EVSE (or CS) has no transactions, we're already in the desired state,
+        // or we're telling the whole CS to power on, we can accept the request - there's nothing stopping us.
         response.status = ChangeAvailabilityStatusEnum::Accepted;
-        // remove any scheduled availability request in case no transaction is scheduled or the component is already in
-        // correct state to override possible requests that have been scheduled before
+        // Remove any scheduled availability requests for the evse_id.
+        // This is relevant in case some of those requests become activated later - the current one overrides them.
         this->scheduled_change_availability_requests.erase(evse_id);
     } else {
-        // add to map of scheduled operational_states. This also overrides successive ChangeAvailability.req with
-        // the same EVSE, which is propably desirable
+        // We can't immediately perform the change, because we have a transaction running.
+        // Schedule the request to run when the transaction finishes.
         this->scheduled_change_availability_requests[evse_id] = {msg, true};
     }
 
-    // send reply before applying changes to EVSE / Connector because this could trigger StatusNotification.req
-    // before responding with ChangeAvailability.req
+    // Respond to the CSMS before performing any changes to avoid StatusNotification.req being sent before
+    // the ChangeAvailabilityResponse.
     ocpp::CallResult<ChangeAvailabilityResponse> call_result(response, call.uniqueId);
     this->send<ChangeAvailabilityResponse>(call_result);
 
     if (!transaction_active) {
-        // execute change availability if possible
-        this->callbacks.change_availability_callback(msg, true);
+        // No transactions - execute the change now
+        this->execute_change_availability_request(msg, true);
```

Comment:
Note: This differs from the original behaviour - in this current state, libocpp will NOT call the callback if the effective state of the components did not change.